docs/samples/deployments/spark/presidio_anonymize_blobs.ipynb

```diff
@@ -0,0 +1,156 @@
+    "from presidio_anonymizer.entities import AnonymizerConfig\n",
+    "from pyspark.sql.types import StringType\n",
+    "from pyspark.sql.functions import input_file_name, regexp_replace\n",
+    "\n",
+    "dbutils.widgets.text(\"storage_account_name\", \"\", \"Blob Storage Account Name\")\n",
+    "dbutils.widgets.text(\"storage_container_name\", \"\", \"Blob Container Name\")\n",
+    "dbutils.widgets.text(\"storage_account_access_key\", \"\", \"Storage Account Access Key\")\n",
+    "dbutils.widgets.text(\"storage_input_folder\", \"input\", \"Input Folder\")\n",
+    "dbutils.widgets.text(\"storage_output_folder\", \"output\", \"Output Folder\")\n"
+   ]
+  },
+  {
+   "source": [
+    "# Import the text files from Azure Blob storage\n"
+   ],
+   "cell_type": "markdown",
+   "metadata": {}
+  },
+  {
+   "source": [
+    "storage_account_name = dbutils.widgets.get(\"storage_account_name\")\n",
+    "storage_container_name = dbutils.widgets.get(\"storage_container_name\")\n",
+    "storage_account_access_key = dbutils.widgets.get(\"storage_account_access_key\")\n",
+    "\n",
+    "\n",
+    "blob_service_client = BlobServiceClient(account_url=\"https://\" + storage_account_name + \".blob.core.windows.net/\", credential=storage_account_access_key)\n",
+    "container_client = blob_service_client.get_container_client(storage_container_name)\n",
+    "\n",
+    "blob_names = container_client.list_blobs(name_starts_with=dbutils.widgets.get(\"storage_input_folder\") + \"/\")\n",
+    "blobs = [] \n",
+    "for blob in blob_names:\n",
+    "  blobs.append(\"wasbs://\" + storage_container_name + \"@\" + storage_account_name + \".blob.core.windows.net/\" + blob.name)\n",
+    "  \n",
+    "spark.conf.set(\"fs.azure.account.key.\"+storage_account_name+\".blob.core.windows.net\", storage_account_access_key)\n",
+    "\n",
+    "input_rdd = spark.read.text(blobs).withColumn(\"filename\", input_file_name())\n",
+    "\n",
+    "input_rdd.show()"
+   ],
+   "cell_type": "code",
+   "metadata": {},
+   "execution_count": null,
+   "outputs": []
+  },
+  {
+   "source": [
+    "# Anonymize Text using Presidio\n"
+   ],
+   "cell_type": "markdown",
+   "metadata": {}
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "def anonymize_text(text):\n",
+    "  analyzer = AnalyzerEngine()\n",
+    "  anonymizer = AnonymizerEngine()\n",
+    "  analyzer_results = analyzer.analyze(text=text, language='en')\n",
+    "  anonymized_results = anonymizer.anonymize(\n",
+    "    text=text,\n",
+    "    analyzer_results=analyzer_results,    \n",
+    "    anonymizers_config={\"DEFAULT\": AnonymizerConfig(\"replace\", {\"new_value\": \"<ANONYMIZED>\"})}\n",
+    "  )\n",
+    "  return anonymized_results\n",
+    "anonymized_rdd = input_rdd.rdd.map(lambda x: (x[\"value\"], x[\"filename\"], anonymize_text(x[\"value\"]))).toDF([\"text\", \"filename\", \"anonymized_text\"])\n",
```

Comment:
Not a spark expert, so I'm not sure what would be the best option. Maybe UDFs or PandasUDFs are going to be faster? I assume they are going to be easier to read which is another consideration here. 
Some references here:
1. https://medium.com/@fqaiser94/udfs-vs-map-vs-custom-spark-native-functions-91ab2c154b44
2. https://medium.com/@paolo_tenti/multi-label-multi-class-batch-classification-in-spark-with-pandas-udfs-f463f1abc09c

Reply:
+1 to what @omri374 said. RDD is a Spark 1.x concept and would be difficult to grasp for someone who started with Spark 2.x. Here are some suggestions,
- When reading the text file we can directly load it into a DataFrame
```python
df = spark.read.csv(blobs, sep=r'\t')
```
- Refactor `anonymize_text()` into a [Pandas UDF](https://databricks.com/blog/2017/10/30/introducing-vectorized-udfs-for-pyspark.html) (recommended over regular Spark UDF due to performance)
- For small text files it probably won't matter but if possible let's reuse the `AnalyzerEngine` and `AnonymizerEngine` instances, otherwise there will be memory allocations on each call of `anonymize_text()`. In case of Pandas UDF, `anonymize_text()` will be called as many number of times as the Spark partitions.
- Instead of `anonymized_rdd.show()`, let's use the built-in Databricks `display(df)`.

Reply:
It makes sense to initialize the analyzer once and broadcast it to the other nodes. It's pretty heavy.

Reply:
done